### PR TITLE
Correctly resolve `&Module.function/arity` syntax

### LINKED ISSUE #532: Entity.resolve ignores arity of captured functions
Hovering over `&Process.delete/1` produces `{:call, Process, :delete, 0}`, but should produce `{:call, Process, :delete, 1}`. This in turn causes find references to fail.

### PR DESCRIPTION
**Pull request by zachallaun**
Fixes #532

### CODE REVIEW

**Review by scohen — APPROVED** [review]

File: apps/remote_control/lib/lexical/remote_control/code_intelligence/entity.ex [hunk 1]
```diff
@@ -272,6 +272,20 @@ defmodule Lexical.RemoteControl.CodeIntelligence.Entity do
     length(args) + 1
   end
 
+  # Calls as part of a capture:
+  # &MyModule.some_function/2
+  defp arity_at_position(
+         [
+           {_, _, []} = call,
```
[the remark this answers is not in the document]
Comment (zachallaun) [reply apps/remote_control/lib/lexical/remote_control/code_intelligence/entity.ex:279]: Yes, I can add a comment to make it more clear -- I'm matching on the a few elements of syntax to ensure we're in a capture with arity: That the innermost call has no args and that it's the first element of a `:/` inside of a `:&`.

[the remark this answers is not in the document]
Comment (zachallaun) [reply apps/remote_control/lib/lexical/remote_control/code_intelligence/entity.ex:279]: Do you think a "diagram" like this is justified?

```elixir
  defp arity_at_position(
         [
           # &MyModule.some_function/2
           #  ^^^^^^^^^^^^^^^^^^^^^^
           {_, _, []} = call,
           # &MyModule.some_function/2
           #  ^^^^^^^^^^^^^^^^^^^^^^^^
           {:/, _, _},
           # &MyModule.some_function/2
           # ^^^^^^^^^^^^^^^^^^^^^^^^^
           {:&, _, [{:/, _, [call, {:__block__, _, [arity]}]}]} | _
         ],
         _position
       )
       when is_call(call) and is_integer(arity) do
    arity
  end
```

[the remark this answers is not in the document]
Comment (zachallaun) [reply apps/remote_control/lib/lexical/remote_control/code_intelligence/entity.ex:279]: And maybe rewriting like this:

```elixir
  defp arity_at_position(
         [
           # &MyModule.some_function/2
           #  ^^^^^^^^^^^^^^^^^^^^^^
           {_, _, []} = call,
           # &MyModule.some_function/2
           #  ^^^^^^^^^^^^^^^^^^^^^^^^
           {:/, _, [call, {:__block__, _, [arity]}]} = slash,
           # &MyModule.some_function/2
           # ^^^^^^^^^^^^^^^^^^^^^^^^^
           {:&, _, [slash]} | _
         ],
         _position
       )
       when is_call(call) and is_integer(arity) do
    arity
  end
```

[the remark this answers is not in the document]
Comment (zachallaun) [reply apps/remote_control/lib/lexical/remote_control/code_intelligence/entity.ex:282]: Went with something a little more explanatory.

**Review by scohen — APPROVED** [review]

**Pull request merged**